cpp/velox/compute/VeloxPlanConverter.cc

```diff
@@ -88,8 +92,39 @@ void VeloxInitializer::Init() {
   }
   auto hdfsPort = hdfsUri.substr(hdfsUri.find(":") + 1);
   auto hdfsHost = hdfsUri.substr(0, hdfsUri.find(":"));
-  static const std::unordered_map<std::string, std::string> configurationValues(
+  static const std::unordered_map<std::string, std::string> hdfsConfig(
       {{"hive.hdfs.host", hdfsHost}, {"hive.hdfs.port", hdfsPort}});
+  configurationValues.merge(hdfsConfig);
+#endif
+
+#ifdef VELOX_ENABLE_S3
+  filesystems::registerS3FileSystem();
+  // TODO(yuan): should passing thru config?
+  std::string awsAccessKey = "minio";
+  std::string awsSecretKey = "miniopass";
+  std::string awsEndpoint = "localhost:9000";
+  const char* envAwsAccessKey = std::getenv("AWS_ACCESS_KEY_ID");
+  if (envAwsAccessKey != nullptr) {
+    awsAccessKey = std::string(envAwsAccessKey);
+  }
+  const char* envAwsSecretKey = std::getenv("AWS_SECRET_ACCESS_KEY");
+  if (envAwsSecretKey != nullptr) {
+    awsSecretKey = std::string(envAwsSecretKey);
+  }
+  const char* envAwsEndpoint = std::getenv("AWS_ENDPOINT");
+  if (envAwsEndpoint != nullptr) {
+    awsEndpoint = std::string(envAwsEndpoint);
+  }
+  std::unordered_map<std::string, std::string> S3Config({
```

Comment:
if ENV variable is the way to pass envAwsAccessKey, envAwsSecretKey and envAwsEndpoint consider returning an error if they are null ?
they have to be set by user, else fail.

Reply:
@akashsha1 right, these configurations should be passed from spark-defaults.conf, will address this in the following commit